velox/connectors/hive/storage_adapters/hdfs/HdfsFileSystem.cpp

```diff
@@ -96,17 +104,86 @@ std::unique_ptr<WriteFile> HdfsFileSystem::openFileForWrite(
   return std::make_unique<HdfsWriteFile>(impl_->hdfsClient(), path);
 }
 
-bool HdfsFileSystem::isHdfsFile(const std::string_view filename) {
-  return filename.find(kScheme) == 0;
+bool HdfsFileSystem::isHdfsFile(const std::string_view filePath) {
+  return filePath.find(kScheme) == 0;
+}
+
+/**
+ * Get hdfs endpoint from config. This is applicable to the case that only one
+ * hdfs endpoint will be used.
+ */
+HdfsServiceEndpoint HdfsFileSystem::getServiceEndpoint(const Config* config) {
+  auto hdfsHost = config->get("hive.hdfs.host");
+  VELOX_CHECK(
+      hdfsHost.hasValue(),
+      "hdfsHost is empty, configuration missing for hdfs host");
+  auto hdfsPort = config->get("hive.hdfs.port");
+  VELOX_CHECK(
+      hdfsPort.hasValue(),
+      "hdfsPort is empty, configuration missing for hdfs port");
+  HdfsServiceEndpoint endpoint{*hdfsHost, *hdfsPort};
+  return endpoint;
+}
+
+/**
+ * Get hdfs endpoint from a given file path, instead of getting a fixed one from
+ * configuration.
+ */
+HdfsServiceEndpoint HdfsFileSystem::getServiceEndpoint(
+    const std::string_view filePath) {
+  auto index1 = filePath.find('/', kScheme.size() + 1);
```

Comment:
minor nit: I would avoid variable names like `index1` and `index2`. Can we name them for what they are? When I read this I have to spend some time figuring out what index1 correlates to. 

Maybe this is more expressive? 
`index1 ` -> `indexOfFirstSlash` /  `beginningOfIdentityInfo` 
`index2` -> `indexOfColon` / ` hostAndPortSeparator` / `beginningOfPortInfo`

What do you think?

Reply:
Thanks for your comment!
Yes, your suggestion makes the code more readable. Just changed them in the new commit.